kiteconnect/connect.py

```diff
@@ -862,7 +876,7 @@ def _request(self, route, method, url_args=None, params=None, is_json=False):
                                         url,
                                         json=params if (method in ["POST", "PUT"] and is_json) else None,
                                         data=params if (method in ["POST", "PUT"] and not is_json) else None,
-                                        params=params if method in ["GET", "DELETE"] else None,
+                                        params=params if method in ["GET", "DELETE"] else query_param,
```

Comment:
This works but it could have been written clearly so you wouldn't have to guess whats happening here 

```python
if method in ["GET", "DELETE"]:
    query_params = params
```

```python
params=query_params
```